src/Simplify/AstHelpers.elm

```diff
@@ -306,6 +367,228 @@ isIdentity lookupTable baseNode =
+    ( ModuleName, String )
+    -> ModuleNameLookupTable
+    -> Node Expression
+    ->
+        Maybe
+            { nodeRange : Range
+            , fnRange : Range
+            , firstArg : Node Expression
+            , argsAfterFirst : List (Node Expression)
+            }
+getSpecificReducedFunctionCall ( moduleName, name ) lookupTable expressionNode =
+    case getSpecificFunctionCall ( moduleName, name ) lookupTable expressionNode of
+        Just call ->
+            Just call
+
+        Nothing ->
+            Maybe.andThen
+                (\reducedLambdaToCall ->
+                    case ( reducedLambdaToCall.lambdaPatterns, reducedLambdaToCall.callArguments ) of
+                        ( [], [] ) ->
+                            Nothing
+
+                        ( _ :: _, [] ) ->
+                            Nothing
+
+                        ( _ :: _, _ :: _ ) ->
+                            Nothing
+
+                        ( [], firstArg :: argsAfterFirst ) ->
+                            Just
+                                { nodeRange = reducedLambdaToCall.nodeRange
+                                , fnRange = reducedLambdaToCall.fnRange
+                                , firstArg = firstArg
+                                , argsAfterFirst = argsAfterFirst
+                                }
+                )
+                (getSpecificReducedLambdaToCall ( moduleName, name ) lookupTable expressionNode)
+
+
+getSpecificReducedLambdaToCall :
+    ( ModuleName, String )
+    -> ModuleNameLookupTable
+    -> Node Expression
+    ->
+        Maybe
+            { nodeRange : Range
+            , fnRange : Range
+            , callArguments : List (Node Expression)
+            , lambdaPatterns : List (Node Pattern)
+            }
+getSpecificReducedLambdaToCall ( moduleName, name ) lookupTable expressionNode =
+    getReducedLambdaToCall expressionNode
+        |> Maybe.andThen
+            (\reducedLambdaToCall ->
+                if
+                    (reducedLambdaToCall.fnName /= name)
+                        || (ModuleNameLookupTable.moduleNameAt lookupTable reducedLambdaToCall.fnRange /= Just moduleName)
+                then
+                    Nothing
+
+                else
+                    Just
+                        { nodeRange = reducedLambdaToCall.nodeRange
+                        , fnRange = reducedLambdaToCall.fnRange
+                        , callArguments = reducedLambdaToCall.callArguments
+                        , lambdaPatterns = reducedLambdaToCall.lambdaPatterns
+                        }
+            )
+
+
+getReducedLambdaToCall :
+    Node Expression
+    ->
+        Maybe
+            { nodeRange : Range
+            , fnName : String
+            , fnRange : Range
+            , callArguments : List (Node Expression)
+            , lambdaPatterns : List (Node Pattern)
+            }
+getReducedLambdaToCall expressionNode =
+    -- maybe a version of this is better located in Normalize?
```

Comment:
I'd say so. also, we might want to have this logic as well in the Normalize logic, so that we notice that `f Err == f (\a -> Err a)`. Let's do this in a separate PR